source/common/http/conn_manager_impl.cc

```diff
@@ -335,6 +340,19 @@ Network::FilterStatus ConnectionManagerImpl::onData(Buffer::Instance& data, bool
   return Network::FilterStatus::StopIteration;
 }
 
+Network::FilterStatus ConnectionManagerImpl::onNewConnection() {
+  if (!read_callbacks_->connection().streamInfo().protocol()) {
+    return Network::FilterStatus::Continue;
+  }
+  // Only QUIC connection's stream_info_ specifies protocol.
+  Buffer::OwnedImpl dummy;
+  codec_ = config_.createCodec(read_callbacks_->connection(), dummy, *this);
```

Comment:
Can you add some more comments in this function about how we are currently injecting QUIC into the HCM? This makes sense to me but it's slightly hacky so it would be good to have a bit of a comment/design trail here.

Reply:
It's still an open question, see https://github.com/envoyproxy/envoy/pull/8537/files#r333005677